sdk/python/feast/client.py

```diff
@@ -508,36 +527,125 @@ def ingest(
         else:
             raise Exception(f"Feature set name must be provided")
 
-        table = _read_table_from_source(source)
+        time_start = time.time()
 
-        # Update the feature set based on DataFrame schema
-        if force_update:
-            # Use a small as reference DataFrame to infer fields
-            ref_df = table.to_batches(max_chunksize=20)[0].to_pandas()
+        # Read table and get row count
+        tmp_table_name = _read_table_from_source(source, chunk_size,
+                                                 max_workers)
+
+        pq_file = pq.ParquetFile(tmp_table_name)
+
+        # Delete source from memory after conversion into PyArrow table
+        del source
+
+        row_count = pq_file.metadata.num_rows
 
-            feature_set.infer_fields_from_df(
-                ref_df, discard_unused_fields=True, replace_existing_features=True
+        # Update the feature set based on PyArrow table of the first row group
+        if force_update:
+            feature_set.infer_fields_from_pa(
+                table=pq_file.read_row_group(0),
+                discard_unused_fields=True,
+                replace_existing_features=True
             )
             self.apply(feature_set)
 
         feature_set = self.get_feature_set(name, version)
 
+        time_end = time.time()
+        exec_time = round(time_end - time_start, 2)
+        print(f"Took {exec_time}s to read, validate and split parquet file.")
```

Comment:
Why does the user care about this?

Reply:
This was meant to be part of my test code, I'll remove this.

Reply:
fixed